### PR TITLE
Skip PhpRedis tests when redis extension is not installed

### PR DESCRIPTION
**Pull request by yatsukhnenko**
Simplest implementation with check whether `redis` extension loaded before acting with `phpredis` property

### CODE REVIEW

File: benchmarks/Support/Benchmark.php [hunk 1]
```diff
@@ -243,16 +247,20 @@ protected function createPredis(): Predis
      */
     public function getBenchmarkMethods(string $filter): array
     {
+        $exclude = null;
+        if (! extension_loaded('redis')) {
+            $exclude = 'PhpRedis';
```
Comment (tillkruss) [inline benchmarks/Support/Benchmark.php:252]: We need to show a warning message when PhpRedis was excluded.

File: benchmarks/Cases/BenchmarkZstdIgbinary.php [hunk 2]
```diff
@@ -57,27 +59,32 @@ public function seed(): void
      */
     protected function setSerialization($client): void
     {
-        $client->setOption(Redis::OPT_SERIALIZER, Redis::SERIALIZER_IGBINARY);
-        $client->setOption(Redis::OPT_COMPRESSION, Redis::COMPRESSION_ZSTD);
+        $client->setOption(Relay::OPT_SERIALIZER, Relay::SERIALIZER_IGBINARY);
+        $client->setOption(Relay::OPT_COMPRESSION, Relay::COMPRESSION_ZSTD);
```
Comment (tillkruss) [inline benchmarks/Cases/BenchmarkZstdIgbinary.php:63]: Can you test this with PhpRedis that was NOT compiled with ZSTD or IGBINARY support to see what happens?

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: benchmarks/Cases/BenchmarkZstdIgbinary.php (hunk 2 above)
Comment (yatsukhnenko) [reply benchmarks/Cases/BenchmarkZstdIgbinary.php:63]: In this case PhpRedis just returns `false`  for `setOption` call

File: benchmarks/Support/Benchmark.php (hunk 1 above)
Comment (yatsukhnenko) [reply benchmarks/Support/Benchmark.php:253]: done

**Comment by yatsukhnenko** [comment]
```bash
root@localhost:/relay# benchmarks/run -h 192.168.1.2 --workers 1 BenchmarkZstdIgbinary.php
Setting up on AMD Ryzen 7 4700U with Radeon Graphics (8 cores/8 threads x86_64)
Using PHP 8.1.21 (OPcache: Off, Xdebug: Off, New Relic: Off)
Connected to Redis (7.0.11) at tcp://192.168.1.2:6379


 WARNING  Failed to set igbinary serializer

 WARNING  Failed to enable zstd compression

 GET (Compressed)  Executing 5 iterations (1 warmup) for 1.00s seconds

+--------------+--------+---------+---------+--------+-------------+---------+--------+--------+
| Client       | Memory | Network | Workers | IOPS   | IOPS/Worker | rstdev  | Change | Factor |
+--------------+--------+---------+---------+--------+-------------+---------+--------+--------+
| Predis       | 3.66mb | 67.54mb |       1 | 17.78K |      17.78K |  ±2.36% |  0.00% |  1.00x |
| RelayNoCache | 3.66mb | 16.45mb |       1 | 18.58K |      18.58K |  ±0.14% |  4.52% |  1.05x |
| PhpRedis     | 3.66mb |  1.19mb |       1 | 29.97K |      29.97K |  ±0.23% | 68.55% |  1.69x |
| Relay        | 3.66mb |    181b |       1 | 52.57K |      52.57K | ±10.08% |   196% |  2.96x |
+--------------+--------+---------+---------+--------+-------------+---------+--------+--------+
```

File: benchmarks/Cases/BenchmarkZstdIgbinary.php (hunk 2 above)
Comment (yatsukhnenko) [reply benchmarks/Cases/BenchmarkZstdIgbinary.php]: done

File: benchmarks/Cases/BenchmarkZstdIgbinary.php [hunk 3]
```diff
@@ -57,27 +60,40 @@ public function seed(): void
...
     }
 
     public function refreshClients(): void
     {
         parent::refreshClients();
-        $this->setSerialization($this->phpredis);
+
+        if (extension_loaded('redis')) {
+            $this->setSerialization($this->phpredis);
+        }
```
Comment (tillkruss) [inline benchmarks/Cases/BenchmarkZstdIgbinary.php:96]: Is the serialization still set for Relay?

File: benchmarks/Support/Benchmark.php [hunk 4]
```diff
@@ -243,16 +249,23 @@ protected function createPredis(): Predis
...
-            function ($method) use ($filter) {
+            function ($method) use ($exclude, $filter) {
                 if (! str_starts_with($method, 'benchmark')) {
                     return false;
                 }
 
                 $method = substr($method, strlen('benchmark'));
 
-                return ! $filter || preg_match("/$filter/i", strtolower($method));
+                return (! $exclude || $method !== $exclude) && (! $filter || preg_match("/$filter/i", strtolower($method)));
```
Comment (tillkruss) [inline benchmarks/Support/Benchmark.php:268]: Can you make this human-readable with a couple of `if` statements?

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: benchmarks/Cases/BenchmarkZstdIgbinary.php (hunk 3 above)
Comment (yatsukhnenko) [reply benchmarks/Cases/BenchmarkZstdIgbinary.php:96]: Serialization is set in `setUpClients` method

File: benchmarks/Support/Benchmark.php (hunk 4 above)
Comment (szepeviktor) [reply benchmarks/Support/Benchmark.php:268]:
```suggestion
                return ($exclude === false || $method !== $exclude)
                    && ($filter === false || preg_match("/{$filter}/i", strtolower($method)));
```

File: benchmarks/Cases/BenchmarkZstdIgbinary.php [hunk 5]
```diff
@@ -57,27 +60,40 @@ public function seed(): void
      */
     protected function setSerialization($client): void
     {
-        $client->setOption(Redis::OPT_SERIALIZER, Redis::SERIALIZER_IGBINARY);
-        $client->setOption(Redis::OPT_COMPRESSION, Redis::COMPRESSION_ZSTD);
+        if (! $client->setOption($client::OPT_SERIALIZER, $client::SERIALIZER_IGBINARY)) {
```
Comment (yatsukhnenko) [inline benchmarks/Cases/BenchmarkZstdIgbinary.php:63]: When `Redis` build without `igbinary` support there is no such constant `SERIALIZER_IGBINARY` and test fails with error
```
Fatal error: Uncaught Error: Undefined constant Redis::SERIALIZER_IGBINARY in /relay/benchmarks/Cases/BenchmarkZstdIgbinary.php:63
```
That's why I used `Relay::SERIALIZER_IGBINARY`

File: benchmarks/Cases/BenchmarkZstdIgbinary.php [hunk 6]
```diff
@@ -57,27 +60,40 @@ public function seed(): void
...
+            Reporter::printWarning(sprintf('Unable to set igbinary serializer on %s', get_class($client)));
+        }
+
+        if (! $client->setOption($client::OPT_COMPRESSION, $client::COMPRESSION_ZSTD)) {
```
Comment (yatsukhnenko) [inline benchmarks/Cases/BenchmarkZstdIgbinary.php:67]: The same problem as described above

File: benchmarks/Support/Benchmark.php (hunk 4 above)
Comment (yatsukhnenko) [reply benchmarks/Support/Benchmark.php]: @szepeviktor `$exclude` initialized with `null` so condition your suggested isn't correct and I've already change it to couple of human-readable if statements :smile:

**Review by tillkruss — APPROVED** [review]
This looks great!

File: benchmarks/Cases/BenchmarkZstdIgbinary.php (hunk 5 above)
Comment (tillkruss) [reply benchmarks/Cases/BenchmarkZstdIgbinary.php]: Ah, should we maybe just do a constant check on the actual class to be explicit about it?

Comment (yatsukhnenko) [reply benchmarks/Cases/BenchmarkZstdIgbinary.php]: done

**Pull request merged**